### DISCUSSION TITLE
Ideas for improvements

### DISCUSSION THREAD

**Discussion by bios4** [devemux86/cruiser#397] [Q&A] [answered]
I'd like to share some of my improvement ideas for navigation and favorites UI...

1. "Simulate navigation" is currently accessible only via some clicks (Menu > Settings > Navigation > Extended > Navigation). My suggestion is to provide a "Simulate navigation" (or "Simulate route") option by long-pressing the routing button (similar to long-press routing button > Stop navigation).

<img src="https://github.com/devemux86/cruiser/assets/145460147/64744c5f-3584-4ba2-94c2-0f33db17bc5b" width="33%"/>

2. "Delete route" is currently also accessible only via some clicks (Menu > Routing > Delete > Overlay/Route > OK). My suggestion is to provide a "Delete route" option by long-pressing the map when a route is loaded.

<img src="https://github.com/devemux86/cruiser/assets/145460147/d25c4693-3896-4d92-88bf-ca94b08b95b6" width="33%"/>

3. When navigating to a set favorite (Menu > My places > Favorites > select favorite > Select on map > Set as destination etc.) I am missing the "back arrow" on that screen to get back to the selected favorite's context menu.

**Comment by devemux86** [comment]
@bios4 thanks for your suggestions.

1. Interesting idea. Question: is the navigation simulation so useful to expose it on the main app buttons?
    - Does anyone use the _static_ simulation or we should remove it?

2. This is already available in route menu: long press _on the route line_, select "Route..." and "Delete"
(the route or any alternatives)

3. Good point, I will see it.

**Reply by bios4** [reply]
> Question: is the navigation simulation so useful to expose it on the main app buttons?

I personally use the "Simulate route" option quite frequently, at least for certain parts of a route.
Especially when being abroad in an unknown (to me) area and the leader of a group of friends, so that there are no surprises on tour.

I never used the "static" simulation so far, so I don't even know what that does... 😉

> This is already available in route menu: long press on the route line

Hmm, this is somehow (very) hidden to me.
Don't you think it makes sense to add this directly accessible and visible to the context menu?

Edit: is it intended that this ^^ option does not ask whether to delete the route, an overlay or both?

**Reply by devemux86** [reply]
> I never used the "static" simulation so far, so I don't even know what that does... 😉

It allows to traverse the route backwards and forwards and listen to the instructions,
without a running location simulation arrow.

> Don't you think it makes sense to add this directly accessible and visible to the context menu?

We can move it one level up in the main route menu (together with the "Recalculate route").

> is it intended that this ^^ option does not ask whether to delete the route, an overlay or both?

I can add a confirmation dialog. It is also used to delete alternative routes,
so the dialog for the overlay should be confusing or not much usable here?

**Comment by devemux86** [comment]
2 + 3 will be available in the next  Cruiser update.

1 is moved to its own discussion as it is a larger feature:
- https://github.com/devemux86/cruiser/discussions/399

<img src="https://github.com/devemux86/cruiser/assets/3484020/934b0ee0-edee-49b9-a6d7-fd5175da34ac" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/c94f7ac4-27e6-4c31-846c-0f7749d237d5" width="50%"/>

**Comment by bios4** [comment]
> 2 + 3 will be available in the next Cruiser update.

Great, thanks!

> We can move it one level up in the main route menu (together with the "Recalculate route").

I suggest to show the "Delete" option AFTER the "Recalculate route" entry and also name it "Delete route".
Just to be exactly clear what is done when clicking it.

**Reply by devemux86** [reply]
> I suggest to show the "Delete" option AFTER the "Recalculate route" entry and also name it "Delete route".
Just to be exactly clear what is done when clicking it.

It will be available in the next Cruiser update.

**Comment by bios4** [comment]
I'd like to share some more things I wanted to see improved @devemux86...

1) "My places" > "Tracks": I am missing the set zoom level for a tracks group/folder in the overview list. In the "Favorites" list, the set zoom level for each list item is displayed (screenshot 1 - green border), but for the "Tracks" list this is missing (screenshot 2 - red border). Could you please add this to the tracks list also!??

<img alt="" width="33%" src="https://github.com/devemux86/cruiser/assets/145460147/2ed87f40-649a-463d-aa9c-477baad5bd9d">
<img alt="" width="33%" src="https://github.com/devemux86/cruiser/assets/145460147/3c3c95b7-e5df-47b0-97b6-b16bd6f385e7">

2) "My places" > "Tracks": I am missing the set color for a tracks group/folder. In the "Favorites" list, the set color for each list item is shown (screenshot 1 above - purple border), but for the "Tracks" list all folder icons are displayed in the set accent color, despite a custom color being set (screenshots 2 above, 3 and 4 - green border). Could you please add this to the tracks list also!??

<img alt="" width="33%" src="https://github.com/devemux86/cruiser/assets/145460147/4783be60-7826-4fee-b02e-fb64aafa50de">
<img alt="" width="33%" src="https://github.com/devemux86/cruiser/assets/145460147/c2b8ebf5-9f30-4b60-bdb7-8aebabf03293">

3) Exporting of tracks to GPX: When a track item contains a slash in the name, the created export GPX of these tracks interprets this slash as a folder structure (screenshots 4 above, 5 and 6 - orange border). It would be great if slashes (and/or backslashes) in the name of favorites, tracks, places etc. are NOT resulting in a folder structure in the export file!

<img alt="" width="33%" src="https://github.com/devemux86/cruiser/assets/145460147/d19171eb-4d0f-4ca6-b477-9b2c75136b8f">
<img alt="" width="33%" src="https://github.com/devemux86/cruiser/assets/145460147/0012efcf-e29a-4401-b7b1-6c3de635fca0">

It would be great if these suggestions could be considered.

**Reply by devemux86** [reply]
@bios4 thanks for the suggestions.

(3) Indeed we should improve this.

---

(1) + (2) Both are because favorite groups have common appearance / visibility.
But routes / tracks can have a separate appearance / visibility per route / track.

For example a group of tracks may contain 2 tracks:
- 1 track set as red and visible from zoom 8
- 1 track set as blue and visible from zoom 10

So their group cannot display a common color or visibility.
We could also have in routes / tracks the color / visibility defined per group, but is it something we want?

Note: also in routes / tracks, we can set the whole group color / visibility in one action.
This is a convenience action, we can still set different color / visibility per route / track.

**Reply by bios4** [reply]
Ah I see...

> We could also have in routes / tracks the color / visibility defined per group, but is it something we want?

I did not even think of setting e.g. the zoom levels different for tracks within the same folder, but for sure this makes sense.
So I would answer your question with "no"... 😁

> (3) Indeed we should improve this.

👍

**Reply by devemux86** [reply]
We could show the color / zoom in the groups, if a group has tracks with same color / zoom.
But it could be confused with groups having the accent color?

**Reply by 0709wiwiwi** [reply]
KISS and watch out that Cruiser app does not become unwieldy and complex due to settings interfering with each other.
Many options also do create setting confusion. An overly complex app deters.
Improvements should above all ensure that simple operation is maintained.

I have also known Locus quite well for some time and the large range of settings is certainly not always favourable.
Although a Locus manual is available, users continue to struggle with the many options offered.
A recent example: [Point labels](https://help.locusmap.eu/topic/33616-point-labels-are-always-shown)

**Reply by devemux86** [reply]
@bios4 

(3) Android by default replaces `/` with `_`, so maybe the OEMs modify Android here.

I included the same replacement in Cruiser, you can try it in:
- https://github.com/devemux86/cruiser/discussions/414

**Reply by bios4** [reply]
Works perfectly fine @devemux86. 👍
Thank you!

**Comment by devemux86** [comment, marked as answer]
All discussed improvements are implemented in [Cruiser 3.0.38](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
